doc/diagnostics/CallingConventions.md

```diff
@@ -0,0 +1,238 @@
+<!--
+Copyright (c) 2020, 2020 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at http://eclipse.org/legal/epl-2.0
+or the Apache License, Version 2.0 which accompanies this distribution
+and is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following Secondary
+Licenses when the conditions for such availability set forth in the
+Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+version 2 with the GNU Classpath Exception [1] and GNU General Public
+License, version 2 with the OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# x86
+
+## x86-64
+
+### Linux
+
+|Register|System Linkage|Caller Preserved|
```

Comment:
"Caller Preserved" here and only "Preserved" in tables below. I'd say we keep it consistent and pick one. Otherwise looks great!

Reply:
For the Z tables, are they Caller or Callee preserved?

Reply:
On Z they are Callee-Preserved registers (AKA non-volatile registers)

Reply:
Just thought of it, is `Caller Preserved` best term to use? those registers are `volatile` or `call-clobbered`

Reply:
🤷 In the internal docs, on x86 they had both caller preserved and callee preserved; from the previous comment thread (https://github.com/eclipse/omr/pull/5254#discussion_r432103490) we decided to remove one of them. The line that the current comment thread is on is on the x86 tables; Filip was asking me to clarify the Z tables (further below) because they only say "Preserved".

Reply:
Requested changes made.